analyzers/tests/SonarAnalyzer.Test/SymbolicExecution/Roslyn/RoslynSymbolicExecutionTest.Invocation.cs

```diff
@@ -1028,6 +1028,21 @@ public void Invocation_ElementOrDefault_CollectionNotEmpty_ReferenceElementType(
         validator.TagValue("Value").Should().HaveNoConstraints();
     }
 
+    [DataTestMethod]
+    [DataRow("FirstOrDefault()")]
+    [DataRow("LastOrDefault()")]
+    [DataRow("SingleOrDefault()")]
+    public void Invocation_ElementOrDefault_CollectionEmpty_ReferenceElementType(string expression)
+    {
+        var code = $"""
+            collection.Clear();
+            var value = collection.{expression};
+            Tag("Value", value);
+            """;
+        var validator = SETestContext.CreateCS(code, $"List<object> collection").Validator;
+        validator.TagValue("Value").Should().HaveOnlyConstraint(ObjectConstraint.Null);
```

Comment:
You can merge these tests into one if you extract the collection type into a parameter:
- `"List<object>"`
- `"Dictionary<int, object>"`

Reply:
yes but I'm testing different things - in one case it's a reference that can return null and in the other case it never returns cause the keyvalue pair is a struct.